pkg/pillar/types/errortime.go

```diff
@@ -34,36 +110,48 @@ func (etPtr *ErrorAndTime) SetError(errStr string, errorTime time.Time) {
 	}
 	etPtr.Error = errStr
 	etPtr.ErrorTime = errorTime
+	etPtr.ErrorSeverity = ErrorSeverityError
+	etPtr.ErrorEntities = []*ErrorEntity{}
+	etPtr.ErrorRetryCondition = ""
+}
+
+// SetErrorWithDescription is when time and description are specified
+// it sets ErrorSeverityError in case of unspecified ErrorSeverity
+// it sets ErrorTime to time.Now() in case of no time provided
+func (etPtr *ErrorAndTime) SetErrorWithDescription(errDescription ErrorDescription) {
+	if errDescription.Error == "" {
+		logrus.Fatal("Missing error string")
+	}
+	etPtr.ErrorDescription = errDescription
+	if etPtr.ErrorSeverity == ErrorSeverityUnspecified {
+		etPtr.ErrorSeverity = ErrorSeverityError
+	}
+	etPtr.ErrorTime = errDescription.ErrorTime
+	if etPtr.ErrorTime.IsZero() {
+		etPtr.ErrorTime = time.Now()
+	}
```

Comment:
This function also needs to copy the Entities and RetryCondition from the errDescription argument, right?

Reply:
It is few lines above (`etPtr.ErrorDescription = errDescription`).

Reply:
So why do you copy ErrorTime again then? That's what confused me